src/components/Rating/Rating.js

```diff
@@ -27,44 +32,62 @@ export default class Rating extends Component {
     const filled = i <= this.props.value
     const hovered = i <= this.state.hoverValue
 
+    const normalColor = this.props.disabled ? {} : {nativeColor: grey[300]};
+    const orangeColor = this.props.disabled ? {} : {nativeColor: orange[500]};
+
+    const iconHovered = !this.props.iconHovered ?
+      <StarBorder
+        classes={{ root: this.props.classes.icon }}
+        {...orangeColor} /> :
+      this.props.iconHovered
+
+    const iconFilled = !this.props.iconFilled ?
+      <Star
+        classes={{root: this.props.classes.icon}}
+        {...orangeColor} /> :
+      this.props.iconFilled
+
+    const iconNormal = !this.props.iconNormal ?
+      <StarBorder
+        className={this.props.classes.icon}
+        {...normalColor} /> :
+      this.props.iconNormal
+
     if ((hovered && !filled) || (!hovered && filled)) {
       return this.props.iconHoveredRenderer ? this.props.iconHoveredRenderer({
         ...this.props,
         index: i
-      }) : this.props.iconHovered
+      }) : iconHovered
     } else if (filled) {
       return this.props.iconFilledRenderer ? this.props.iconFilledRenderer({
         ...this.props,
         index: i
-      }) : this.props.iconFilled
+      }) : iconFilled
     } else {
       return this.props.iconNormalRenderer ? this.props.iconNormalRenderer({
         ...this.props,
         index: i
-      }) : this.props.iconNormal
+      }) : iconNormal
     }
   }
 
   render () {
     const rating = []
 
     for (let i = 1; i <= this.props.max; i++) {
-      const tooltip = this.props.tooltip || this.props.tooltipRenderer ? this.props.tooltipRenderer({index: i, ...this.props}) : null
-
       rating.push(
         <IconButton
           key={i}
-          className={this.props.className}
+          className={classNames(
+            this.props.classes.iconButton,
+            {
+              [this.props.classes.disabled]: this.props.disabled || this.props.readOnly
```

Comment:
I see that this was ported from the previous approach, but semantically, using the same `disabled` class for two states (`readonly` vs. `disabled`) doesn't make much sense. Maybe they should be splitted so that the user can customize the styles for both cases individually.

Reply:
Done